src/multi-select/multi-select.component.jsx

```diff
@@ -16,31 +17,47 @@ export default class MultiSelect extends React.PureComponent {
       text: PropTypes.string.isRequired,
     })).isRequired,
     checkedItems: ImmutablePropTypes.map,
+    maxHeight: PropTypes.number,
     onChange: PropTypes.func,
   };
 
   static defaultProps = {
     checkedItems: Map(),
+    maxHeight: 400,
     onChange: () => {},
   };
 
+  handleChange = (id, isChecked) => {
+    const { checkedItems, onChange } = this.props;
+    if (isChecked) {
+      onChange(checkedItems.set(id, isChecked));
+    } else {
+      onChange(checkedItems.delete(id));
+    }
+  }
+
   isChecked = (id, checkedItems) => checkedItems.get(id) === true;
 
   render() {
-    const { items, checkedItems, onChange } = this.props;
+    const { items, checkedItems, maxHeight } = this.props;
+    const ITEM_HEIGHT = 30;
+    const ITEM_MARGIN = 10;
+    const height = (ITEM_HEIGHT * items.length) + ITEM_MARGIN;
     return (
-      <div className="oc-multi-select">
-        {items.map((item) => {
-          const isChecked = this.isChecked(item.id, checkedItems);
-          return (
-            <MultiSelectItem
-              key={item.id}
-              isChecked={isChecked}
-              item={item}
-              onChange={onChange}
-            />
-          );
-        })}
+      <div className="oc-multi-select" style={{ height, maxHeight }}>
```

Comment:
Do you mean overflow-y: scroll (or auto); for the container? If yes, there will be a default scrollbar